Frontend/HsTypes.hs

```diff
@@ -285,6 +277,40 @@ isHsArrowTy (TyApp (TyApp (TyCon tc) ty1) ty2)
   | isArrowTyCon tc   = Just (ty1, ty2)
 isHsArrowTy _other_ty = Nothing
 
+-- * Unit Data and Type Con
+-- ------------------------------------------------------------------------------
+
+unitSym :: Sym
+unitSym = mkSym "()"
+
+unitName :: Name
+unitName = mkName unitSym unitUnique
+
+psUnitTyCon :: PsTyCon
+psUnitTyCon = HsTC unitSym
+
+rnUnitTyCon :: RnTyCon
+rnUnitTyCon = HsTC unitName
+
+unitTyConInfo :: HsTyConInfo
+unitTyConInfo = HsTCInfo rnUnitTyCon
+                         []
+                         fcUnitTyCon
+                         [rnUnitDataCon]
+
+psUnitDataCon :: PsDataCon
+psUnitDataCon = HsDC unitSym
+
+rnUnitDataCon :: RnDataCon
+rnUnitDataCon = HsDC unitName
```

Comment:
I would prefer it if the data constructor and the type constructor used different uniques. Instead of a single `unitUnique` I think it is better to have a `unitTyConUnique` and `unitDataConUnique`.